core/src/test/java/org/apache/calcite/test/SpatialTypeTableFunctionsTest.java

```diff
@@ -0,0 +1,106 @@
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test;
+
+import org.apache.calcite.jdbc.CalciteConnection;
+import org.apache.calcite.model.ModelHandler;
+import org.apache.calcite.runtime.SpatialTypeFunctions;
+import org.apache.calcite.runtime.SpatialTypeTableFunctions;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.schema.TableFunction;
+import org.apache.calcite.schema.impl.TableFunctionImpl;
+import org.apache.calcite.sql.fun.SqlSpatialTypeFunctions;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Tests for spatial table functions.
+ */
+public class SpatialTypeTableFunctionsTest {
+
+  /**
+   * Tests the ST_Explode table function.
+   */
+  @Test void testExplodeFunction() throws SQLException {
+    try (Connection connection = DriverManager.getConnection("jdbc:calcite:")) {
+      CalciteConnection calciteConnection =
+          connection.unwrap(CalciteConnection.class);
+      SchemaPlus rootSchema = calciteConnection.getRootSchema();
+      final ImmutableList<String> emptyPath = ImmutableList.of();
+      ModelHandler.addFunctions(rootSchema, null, emptyPath,
+          SpatialTypeFunctions.class.getName(), "*", true);
+      ModelHandler.addFunctions(rootSchema, null, emptyPath,
+          SqlSpatialTypeFunctions.class.getName(), "*", true);
+      final TableFunction explode = TableFunctionImpl
+          .create(SpatialTypeTableFunctions.EXPLODE);
+      rootSchema.add("ST_EXPLODE", explode);
+      final String sql = "SELECT * FROM table(ST_Explode("
+          + "ST_GeomFromText('MULTIPOINT((1 1), (2 2), (3 3), (4 4))')"
+          + "))";
+      ResultSet resultSet = connection.createStatement().executeQuery(sql);
+      assertThat(CalciteAssert.toString(resultSet),
+          equalTo("geom=POINT (1 1)\n"
+              + "geom=POINT (2 2)\n"
+              + "geom=POINT (3 3)\n"
+              + "geom=POINT (4 4)\n"));
+    }
+  }
+
+  /**
+   * Tests the ST_EXPLODECURSOR table function.
+   */
+  @Disabled()
+  @Test void testExplodeCursorFunction() throws SQLException {
```

Comment:
Is it a goal to implement `ST_ExploreCursor` in this change? If so this test should be enabled.

Reply:
The EXPLODE_CURSOR function is currently not working and the corresponding examples in `TableFunctionTest.testTableFunctionCursorInputs` neither (the original test is disabled as well). Is there another way to achieve the same result?

Reply:
I don't recall what is and isn't working with table functions. Best thing to do is to log a Jira case with what's not working, and make [CALCITE-5281] depend on it.

Reply:
Thanks, i logged [CALCITE-5309](https://issues.apache.org/jira/browse/CALCITE-5309) and will continue the discussion in [CALCITE-5281](https://issues.apache.org/jira/browse/CALCITE-5281).

Reply:
I removed the cursor functions for now and tried to clarify the proble in [CALCITE-5281](https://issues.apache.org/jira/browse/CALCITE-5281).